src/f32-qs8-vcvt/rvv.c.in

```diff
@@ -0,0 +1,55 @@
+// Copyright 2024 Imagination Technologies, inc.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+$assert LMUL in [1, 2, 4, 8]
+$LMUL_8 =  {1: "f4", 2: "f2", 4: "1", 8: "2"}[LMUL]
+$LMUL_16 = {1: "f2", 2: "1",  4: "2", 8: "4"}[LMUL]
+$XINT8_T = {"QS8": "int8_t", "QU8": "uint8_t"}[DATATYPE]
+#include <assert.h>
+
+#include <riscv_vector.h>
+#include "xnnpack/math.h"
+#include "xnnpack/vcvt.h"
+
+
+void xnn_f32_${DATATYPE.lower()}_vcvt_ukernel__rvv_u${LMUL}v(
+    size_t batch,
+    const float* input,
+    ${XINT8_T}* output,
+    const union xnn_f32_${DATATYPE.lower()}_cvt_params params[restrict XNN_MIN_ELEMENTS(1)]) XNN_OOB_READS
+{
+  assert(batch != 0);
+  assert(batch % sizeof(float) == 0);
+  assert(input != NULL);
+  assert(output != NULL);
+
+  batch >>= XNN_LOG2_SIZEOF_FLOAT;
+
+  const float scale = params->scalar.scale;
+  const vfloat32m${LMUL}_t magic_bias_f32v = __riscv_vfmv_v_f_f32m${LMUL}(12582912.0f, __riscv_vsetvl_e32m${LMUL}(batch));
+  const float output_min_less_zero_point = (float) ((int32_t) params->scalar.output_min - (int32_t) params->scalar.output_zero_point);
+  const float output_max_less_zero_point = (float) ((int32_t) params->scalar.output_max - (int32_t) params->scalar.output_zero_point);
+  const int32_t magic_min = (int32_t) float_as_uint32(12582912.0f + output_min_less_zero_point);
```

Comment:
is the 'magic' method really the fastest on riscv?

The basic function would load floats, scale, bias, clamp and convert to bytes.
the clamping can be done as bytes if it is faster.

Also I think it is common to not require clamping, so we may want to consider a 'linear' version.
In the past we would also consider doing multiple versions - e.g. fmagic as a 'variant' or different template, incase some hardware prefers one way or the other.

Reply:
The magic implementation saves one vector instruction, and doesn't utilize as many converts, if you prefer one variant over the other let me know which you would like me to go forward with. I can do multiple versions if you deem it necessary also.